packages/rum/src/domain/record/record.ts

```diff
@@ -69,50 +67,38 @@ export function record(options: RecordOptions): RecordAPI {
 
   function flushMutations() {
     shadowRootsController.flush()
-    flushMutationsFromObservers()
+    mutationTracker.flush()
   }
 
-  const { stop: stopObservers, flush: flushMutationsFromObservers } = initObservers(configuration, {
-    lifeCycle: options.lifeCycle,
+  const recordIds = initRecordIds()
+  const mutationTracker: FlushableTracker = trackMutation(
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Do we need this `FlushableTracker` abstraction? it seems to be used only for `MutationTracker`

Reply:
Following @thomas-lebeau reasoning in his [comment](https://github.com/DataDog/browser-sdk/pull/2659#discussion_r1533720875): the types are inferred quite well, but being more explicit could help reading and understanding the code.
So if we strongly types other trackers, I would expect to also have a type here. Maybe instead of a `FlushableTracker` concept it could be `type MutationTracker = Tracker & { flush: () => void }` collocated within `trackMutation.ts`. Wdyt?